plugin/adsk/scripts/AETemplateHelpers.py

```diff
@@ -11,6 +12,46 @@ def debugMessage(msg):
     if DEBUG:
         print(msg)
 
+def GetAllRootPrimNamesNaturalOrder(proxyShape):
+    # Custom comparator
+    def naturalOrderComparator(a, b):
```

Comment:
I would move this function out. I would also rewrite it like this (You can run `python -m doctest` on this to verify it is working). My version handle case where the numbers are 1 and 01 which are equal but not the same.

```
def _splitInParts(a):
    '''
    Convert a string into a tuple of parts, where each part is either a string
    without digits or a tuple containing a number and the length of its representation.

    >>> _splitInParts('a')
    ('a',)
    >>> _splitInParts('a1')
    ('a', (1, 1))
    >>> _splitInParts('a01b')
    ('a', (1, 2), 'b')
    >>> _splitInParts('120')
    ((120, 3),)
    '''
    parts = []
    part = None
    for c in a:
        if c.isdigit():
            if type(part) is str:
                parts.append(part)
                part = None
            if part is None:
                part = (int(c), 1)
            else:
                part = (part[0] * 10 + int(c), part[1] + 1)
        else:
            if type(part) is tuple:
                parts.append(part)
                part = None
            if part is None:
                part = c
            else:
                part = part + c
    if part is not None:
        parts.append(part)
    return tuple(parts)


def stringNumCompare(a, b):
    '''
    Compare two strings, but compare embedded numbers as numbers.
    >>> stringNumCompare('a', 'b')
    -1
    >>> stringNumCompare('b', 'a')
    1
    >>> stringNumCompare('abcd', 'abcd')
    0
    >>> stringNumCompare('10', '9')
    1
    >>> stringNumCompare('7', '12')
    -1
    >>> stringNumCompare('10a', '9b')
    1
    >>> stringNumCompare('7f', '12cde')
    -1
    >>> stringNumCompare('abcd10', 'abcd1')
    1
    >>> stringNumCompare('abcd01', 'abcd01')
    0
    >>> stringNumCompare('abcd1', 'abcd01')
    -1
    >>> stringNumCompare('abcd3', 'abcd1')
    1
    >>> stringNumCompare('abcd1', 'abcd5')
    -1
    '''
    a_parts = _splitInParts(a)
    b_parts = _splitInParts(b)
    if a_parts < b_parts:
        return -1
    elif a_parts > b_parts:
        return 1
    else:
        return 0
```